### PR TITLE
EZP-27737: Load correct groups/roles in ezuser

### PR DESCRIPTION
**Pull request by wizhippo**
> https://jira.ez.no/browse/EZP-27737

The fixes a regression after https://github.com/ezsystems/ezpublish-legacy/commit/134b652e66a399743ba158bf06c101c8abca7bd4

### CODE REVIEW

File: kernel/classes/datatypes/ezuser/ezuser.php [hunk 1]
```diff
@@ -1423,8 +1423,8 @@ private static function generateUserCacheData( $userId )
         $data['groups'] = $user->groups();
 
         // role list (session: eZRoleIDList)
-        $groups[] = $userId;
-        $data['roles'] = eZRole::fetchIDListByUser( $groups );
+        $data['groups'][] = $userId;
```
Comment (adamwojs) [inline kernel/classes/datatypes/ezuser/ezuser.php]: This method returns `$data` as result. In original behaviour  `$data['groups']` wasn't contained `$userId` at the end. So I think that this code should looks like as follows:

```php
$groups = $user->groups(); 
$data['groups'] = $groups;
$groups[] = $userId;

$data['roles'] = eZRole::fetchIDListByUser($groups);
```

**Comment by wizhippo** [comment]
Not sure why eZDFSClusterStaleCacheTest::testStaleCache is failing.  Don't think is has anything to do with this commit.

**Comment by adamwojs** [comment]
> Don't think is has anything to do with this commit.

@wizhippo Other PR are also failing this test, so don't worry about this 😉

**Comment by andrerom** [comment]
Merged, thanks @wizhippo!

**Pull request merged**